mantle/kola/tests/misc/network.go

```diff
@@ -219,6 +229,81 @@ var (
           umount /var/mnt
 	`
 
+	setupVethPairsTemplate = `#!/usr/bin/env bash
+          set -ex
+
+          create_veth_pair() {
+            veth_host_side_end_name=$1
+            veth_netns_side_end_name=$2
+            host_side_mac_address=$3
+            netns_side_ip=$4
+            network_namespace=$5
+
+            # Create veth pair and assign a namespace to veth-netns
+            ip link add ${veth_host_side_end_name} type veth peer name ${veth_netns_side_end_name}
+            ip link set ${veth_netns_side_end_name} netns ${network_namespace}
+
+            # Assign an MAC address to the host-side veth end
+            ip link set dev ${veth_host_side_end_name} address ${host_side_mac_address}
+
+            # Assign an IP address to netns-side veth end and bring it up
+            ip netns exec ${network_namespace} ip address add ${netns_side_ip} dev ${veth_netns_side_end_name}
+            ip netns exec ${network_namespace} ip link set ${veth_netns_side_end_name} up
+          }
+
+          activate_veth_end() {
+            veth_end_name=$1
+
+            nmcli dev set ${veth_end_name} managed yes
+            ip link set ${veth_end_name} up
+
+            poll_dhcp_success ${veth_end_name}
+          }
+
+          poll_dhcp_success() {
+            veth_end_name=$1
+            for attempt in {1..5}; do
+              cidr=$(nmcli -g IP4.ADDRESS dev show ${veth_end_name})
+              if [[ ! -z ${cidr} ]]; then
```

Comment:
This will fail if the output is empty. You have to put quotes around the variable

Reply:
Done